### PR TITLE
Match EndActorSpeech

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

File: src/181810.c [hunk 1]
```diff
@@ -108,7 +108,72 @@ ApiStatus ActorSpeak(Evt* script, s32 isInitialCall) {
...
+        gSpeakingActor = actor;
+        gSpeakingActorPart = actorPart;
+        close_message(gSpeakingActorPrintCtx);
+        script->functionTemp[0] = 0;
+        increment_status_menu_disabled();
+    }
+
+    if (script->functionTemp[0] == 0) {
+        Actor* actor = gSpeakingActor;
+        ActorPart *actorPart = gSpeakingActorPart;
```
Comment (ethteck) [inline src/181810.c:140]: `ActorPart*`

**Comment by JoshDuMan** [comment]
https://decomp.me/scratch/Ctruw

File: src/181810.c [hunk 2]
```diff
@@ -108,7 +108,72 @@ ApiStatus ActorSpeak(Evt* script, s32 isInitialCall) {
...
+
+        x = actor->currentPos.x + actor->headOffset.x;
+        if ((gSpeakingActor->flags & 0x8000) == 0) {
```
Comment (ethteck) [inline src/181810.c:143]: `!(gSpeakingActor->flags...`

**Review by ethteck — CHANGES_REQUESTED** [review]
Great match!

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.02%) on `us`.

⚠️ This PR introduces a warning:
- src/181810.c:133: warning: implicit declaration of function `close_message'
- src/181810.c:350: warning: implicit declaration of function `show_foreground_models_unchecked'
- src/181810.c:352: warning: implicit declaration of function `hide_foreground_models_unchecked'
- src/181810.c:359: warning: implicit declaration of function `show_foreground_models'
- src/181810.c:361: warning: implicit declaration of function `hide_foreground_models'
- src/181810.c:420: warning: implicit declaration of function `btl_set_state'
- src/181810.c:468: warning: implicit declaration of function `start_rumble_type'

**Review by ethteck — APPROVED** [review]

File: src/181810.c [hunk 3]
```diff
@@ -108,7 +108,72 @@ ApiStatus ActorSpeak(Evt* script, s32 isInitialCall) {
...
-INCLUDE_ASM(s32, "181810", EndActorSpeech);
+ApiStatus EndActorSpeech(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 flags;
+    s32 anim;
+    f32 x, y, z;
+    s32 screenX, screenY, screenZ;
+
+    if (isInitialCall) {
+        Actor* actor = (Actor*) evt_get_variable(script, *args++);
```
Comment (nanaian) [inline src/181810.c:119]: This should be typed s32, get_actor(id) returns an Actor*

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/181810.c (hunk 3 above)
Comment (ethteck) [reply src/181810.c:119]: He had to re-use the same var for the id and the actor itself, so it's a cast one way or the other

**Pull request merged**